src/sidebar/components/tag-editor.js

```diff
@@ -1,31 +1,38 @@
 import { createElement } from 'preact';
-import { useMemo, useRef, useState } from 'preact/hooks';
+import { useRef, useState } from 'preact/hooks';
 import propTypes from 'prop-types';
 
 import { withServices } from '../util/service-context';
 
+import Datalist from './datalist';
 import SvgIcon from './svg-icon';
+import useElementShouldClose from './hooks/use-element-should-close';
 
 // Global counter used to create a unique id for each instance of a TagEditor
-let datalistIdCounter = 0;
+let tagEditorIdCounter = 0;
 
 /**
  * Component to edit annotation's tags.
+ *
+ * Component accessibility is modeled after "Combobox with Listbox Popup Examples" found here:
+ * https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html
```

Comment:
👍 - Let's make sure we add similar comments on any other complex widgets we implement from these guidelines.

Reply:
Good idea!